### PR TITLE
Added a focus back to the editor, when options from the settings panel are used

### PR DESCRIPTION
**Pull request by sagzy**
refs https://github.com/TryGhost/Team/issues/3217

- kudos to Kevin for the idea
- demo:

https://github.com/TryGhost/Koenig/assets/6225080/8778caf6-4122-4a43-9a1c-22937884a805

### CODE REVIEW

File: packages/koenig-lexical/src/hooks/usePreviousFocus.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import {useState} from 'react';
+
+export const usePreviousFocus = (onClick, name) => {
+    const [previousRange, setPreviousRange] = useState(null);
```
Comment (binary-koan) [inline packages/koenig-lexical/src/hooks/usePreviousFocus.js:4]: I would make this a ref instead of state to avoid extra re-renders

Comment (sagzy) [reply packages/koenig-lexical/src/hooks/usePreviousFocus.js]: great point, updated in https://github.com/TryGhost/Koenig/pull/680/commits/495aa25105f34bbb5f952aeb37ec31f61ac08da3

**Review by binary-koan — APPROVED** [review]
Looks good and works well for different cursor positions, selected text, etc - nice!

**Review by kevinansfield — APPROVED** [review]

**Comment by kevinansfield** [comment]
Looks good! Would be nice to get a test for the selection behaviour in case we change/refactor anything later down the line

**Comment by sagzy** [comment]
> Looks good! Would be nice to get a test for the selection behaviour in case we change/refactor anything later down the line

Yes def., added some specs in https://github.com/TryGhost/Koenig/pull/680/commits/d401ede974f9b7d37b20a5ea6cf634621a8eb17b – will merge after the CI is ✅

**Pull request merged**